cirq-core/cirq/sim/simulation_state_test.py

```diff
@@ -101,3 +161,75 @@ def test_field_getters():
     args = DummySimulationState()
     assert args.prng is np.random
     assert args.qubit_map == {q: i for i, q in enumerate(cirq.LineQubit.range(2))}
+
+
+@pytest.mark.parametrize('exp', [-3, -2, -1, 0, 1, 2, 3])
+def test_ancillaZ(exp):
+    q = cirq.LineQubit(0)
+    test_circuit = cirq.Circuit(AncillaZ(exp).on(q))
+
+    control_circuit = cirq.Circuit(cirq.ZPowGate(exponent=exp).on(q))
+
+    test_sv = cirq.final_state_vector(test_circuit)
+    control_sv = cirq.final_state_vector(control_circuit)
+    assert np.allclose(test_sv, control_sv)
+
+    test_dm = cirq.final_density_matrix(test_circuit)
```

Comment:
This repeats the above three lines. You could parameterize a test functor here instead.